File: java/oop/CollegeApp/src/com/techlabs/college/Student.java
package com.techlabs.college;

import java.text.ParseException;

public class Student extends Person {

	private Branch branch;
	private int semester;
	
	public Student(int id,String name,String address,String dob,Branch branch,int semester) throws ParseException{
		super(id, name, address, dob);
		this.branch=branch;
		this.semester=semester;
	}
	
	public String getBranch(){
		return this.branch.name();
	}
	
	public int getSemester(){
		return this.semester;
	}
}

File: dotNet/CSharp/CollegeApp/CollegeApp/Student.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CollegeApp
{
    class Student : Person
    {
        private Branch _branch;
        private int _semester;
        public Student(int id, string name, string address, string dateOfBirth,Branch branch,int semester) : base(id, name, address, dateOfBirth)
        {
            _branch = branch;
            _semester = semester;
        }

        public String GetBranch
        {
            get
            {
                return _branch.ToString();
            }
        }

        public Int32 Semester
        {
            get
            {
                return _semester;
            }
        }

        public override string ToString()
        {
            return "Id: "+Id
                +"\n Name: "+Name
                +"\n Address: "+Address
                +"\n DateOfBirth: "+DateOfBirth
                +"\n Branch: "+GetBranch
                +"\n Semester: "+Semester;
        }
    }
}
